### PR TITLE
add firestore and upgrade to 0.30.0

### LINKED ISSUE #498: Update libraries.json to 0.30.0
released products are now at 1.12.0 and vision is now GA

### PR DESCRIPTION
**Pull request by elharo**
@etanshaul @loosebazooka fix #498

### CODE REVIEW

File: src/main/resources/com/google/cloud/tools/libraries/libraries.json [hunk 1]
```diff
@@ -200,7 +200,31 @@
...
-         "version": "0.29.0-beta"
+         "version": "0.30.0-beta"
+       }
+      }
+    ]
+  },
+  {
+    "name": "Google Cloud Firestore",
+    "id": "cloudfirestore",
+    "serviceName": "firestore.googleapis.com",
```
Comment (etanshaul) [inline src/main/resources/com/google/cloud/tools/libraries/libraries.json]: Did you confirm that no roles are associated with this service?

Comment (etanshaul) [reply src/main/resources/com/google/cloud/tools/libraries/libraries.json]: (im trying to check this myself too)

Comment (elharo) [reply src/main/resources/com/google/cloud/tools/libraries/libraries.json]: I can't find any documentation of any roles anywhere. Where should I look?

Comment (etanshaul) [reply src/main/resources/com/google/cloud/tools/libraries/libraries.json]: I didn't find any. You can get a list of all available roles from the Iam api:

`iam.roles().list()`

approving

Comment (etanshaul) [reply src/main/resources/com/google/cloud/tools/libraries/libraries.json]: fyi its also available via gcloud:

https://cloud.google.com/sdk/gcloud/reference/iam/list-grantable-roles

**Pull request merged**